Gems/ROS2/Code/Source/RobotImporter/RobotImporterWidget.cpp

```diff
@@ -0,0 +1,318 @@
+            int ret = msgBox.exec();
+            if (ret == QMessageBox::Cancel)
+            {
+                m_prefabMakerPage->setSuccess(false);
+                return;
+            }
+        }
+        m_prefabMaker = AZStd::make_unique<URDFPrefabMaker>(m_urdfPath, m_parsedUrdf, prefabPath.String(), m_urdfAssetsMapping);
+
+        auto callback = [&]()
+        {
+            emit SignalFinalizeURDFCreation();
+        };
+        m_prefabMaker->LoadURDF(callback);
+    }
+    void RobotImporterWidget::FinalizeURDFCreation()
+    {
+        auto prefabOutcome = m_prefabMaker->CreatePrefabFromURDF();
+        if (prefabOutcome.IsSuccess())
+        {
+            AZStd::string status = m_prefabMaker->getStatus();
+            m_prefabMakerPage->reportProgress(status);
+            m_prefabMakerPage->setSuccess(true);
+        }
+        else
+        {
+            AZStd::string status = "Failed to create prefab\n";
+            status += prefabOutcome.GetError() + "\n";
+            status += m_prefabMaker->getStatus();
+            m_prefabMakerPage->reportProgress(status);
+            m_prefabMakerPage->setSuccess(false);
+        }
+    }
+    void RobotImporterWidget::onCreateButtonPressed()
+    {
+        CreatePrefab(m_prefabMakerPage->getPrefabName());
+    }
+
+    bool RobotImporterWidget::CheckCyclicalDependency(AZ::IO::Path importedPrefabPath)
+    {
+        AzFramework::EntityContextId context_id;
+        EBUS_EVENT_RESULT(context_id, AzFramework::EntityIdContextQueryBus, GetOwningContextId);
+
+        AZ_Printf("CheckCyclicalDependency", "CheckCyclicalDependency %s", importedPrefabPath.Native().data());
+        auto focus_interface = AZ::Interface<AzToolsFramework::Prefab::PrefabFocusInterface>::Get();
+
+        if (!focus_interface)
+        {
+            ReportError(tr("Imported prefab could not be validated.\nImport aborted."));
+            return true;
+        }
+
+        auto focus_prefab_instance = focus_interface->GetFocusedPrefabInstance(context_id);
+
+        if (!focus_prefab_instance)
+        {
+            ReportError(tr("Imported prefab could not be validated.\nImport aborted."));
+            return true;
+        }
+
+        auto focus_prefab_filename = focus_prefab_instance.value().get().GetTemplateSourcePath();
+
+        AZ_Printf("CheckCyclicalDependency", "focus_prefab_filename %s", focus_prefab_filename.Native().data());
+        if (focus_prefab_filename == importedPrefabPath)
+        {
+            ReportError(
+                tr("Cyclical dependency detected.\nSelected URDF model is currently being edited. Exit prefab edit mode and try again."));
+            return true;
+        }
+
+        return false;
+    }
+
+    void RobotImporterWidget::ReportError(const QString& errorMessage)
+    {
+        QMessageBox::critical(this, QObject::tr("Error"), errorMessage);
+        AZ_Error("RobotImporterWidget", false, errorMessage.toStdString().c_str());
```

Comment:
toUtf8().constData() should be used over `toStdString` for converting QT UTF16 string data to UtF8
```suggestion
        AZ_Error("RobotImporterWidget", false, "%s", errorMessage.toUtf8().constData());
```